src/push-manifest.ts

```diff
@@ -41,72 +81,106 @@ const extensionToTypeMapping: {[extname: string]: LinkType} = {
   '.woff': 'font',
   '.woff2': 'font'
 };
-export interface PushManifestEntry {
-  [dependencyAbsoluteUrl: string]: {
-    type?: LinkType,
-    // NOTE(fks) 04-05-2017: Only weight=1 is supported by browsers at the
-    // moment. When support is added, we can add automatic weighting and support
-    // multiple numbers.
+
+/**
+ * Get the default resource type for a file based on its extension.
+ */
+function getResourceTypeFromUrl(url: string): ResourceType|undefined {
+  return extensionToTypeMapping[path.extname(url)] || undefined;
+}
+
+/**
+ * Get the resource type for an import, handling special import types and
+ * falling back to getResourceTypeFromUrl() if none exist.
+ */
+function getResourceTypeFromImport(importFeature: Import): ResourceType|
+    undefined {
+  const importTagName = importFeature.astNode.tagName;
```

Comment:
I was about to file an issue about that in the analyzer :) The type of language/astNode format is something that I'd like to check before accessing it. Checking the existence of the `tagName` property was the quick and dirty solution I came up with, but I like checking `kinds` much better.